server/consumer.go

```diff
@@ -544,6 +547,14 @@ func checkConsumerCfg(
 		}
 	}
 
+	var metadataLen int
+	for k, v := range config.Metadata {
+		metadataLen += len(k) + len(v)
+	}
+	if metadataLen > JSMaxMetadataLen {
+		return NewJSConsumerMetadataLengthError(fmt.Sprintf("%dKB", JSMaxMetadataLen/1024))
```

Comment:
check for empty keys?

Reply:
I prefer to be as transparent as possible on the server side and add sanity checks on the client side (according to client).

Reply:
empty keys are persisted but seems benign to support that

Reply:
yeah, I think we can leave it as it is.